diff --git a/mars_uav_sim/perfect_drone_sim/ros/ros2.CMakeLists.txt b/mars_uav_sim/perfect_drone_sim/ros/ros2.CMakeLists.txt
index 2518dae..7569c33 100755
--- a/mars_uav_sim/perfect_drone_sim/ros/ros2.CMakeLists.txt
+++ b/mars_uav_sim/perfect_drone_sim/ros/ros2.CMakeLists.txt
@@ -31,6 +31,7 @@ find_package(visualization_msgs REQUIRED)
 find_package(marsim_render REQUIRED)
 find_package(mars_quadrotor_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
+find_package(pcl_conversions REQUIRED)
 
 # Other packages
 find_package(Eigen3 REQUIRED)
@@ -63,6 +64,7 @@ set(dependencies
         nav_msgs
         visualization_msgs
         tf2_ros
+        pcl_conversions
 )
 
 add_executable(perfect_drone_node
diff --git a/mars_uav_sim/perfect_drone_sim/ros/ros2.package.xml b/mars_uav_sim/perfect_drone_sim/ros/ros2.package.xml
index 2114f39..e40ebe0 100755
--- a/mars_uav_sim/perfect_drone_sim/ros/ros2.package.xml
+++ b/mars_uav_sim/perfect_drone_sim/ros/ros2.package.xml
@@ -12,6 +12,7 @@
     <depend>std_msgs</depend>
     <depend>mars_quadrotor_msgs</depend>
     <depend>marsim_render</depend>
+    <depend>pcl_conversions</depend>
     <export>
         <build_type>ament_cmake</build_type>
     </export>
